packages/kbn-expandable-flyout/src/components/render_menu.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  RENDER_MENU_BUTTON_TEST_ID,
+  RENDER_MENU_FLYOUT_TYPE_BUTTON_GROUP_OVERLAY_TEST_ID,
+  RENDER_MENU_FLYOUT_TYPE_BUTTON_GROUP_PUSH_TEST_ID,
+  RENDER_MENU_FLYOUT_TYPE_BUTTON_GROUP_TEST_ID,
+  RENDER_MENU_FLYOUT_TYPE_INFORMATION_ICON_TEST_ID,
+  RENDER_MENU_FLYOUT_TYPE_TITLE_TEST_ID,
+} from './test_ids';
+
+const RENDER_MENU_ICON_BUTTON = i18n.translate('expandableFlyout.renderMenu.popoverButton', {
+  defaultMessage: 'Open flyout render menu',
+});
+const FLYOUT_TYPE_TITLE = i18n.translate('expandableFlyout.renderMenu.flyoutTypeTitle', {
+  defaultMessage: 'Flyout type',
+});
+const FLYOUT_TYPE_OVERLAY_MODE = i18n.translate('expandableFlyout.renderMenu.overlayMode', {
+  defaultMessage: 'Overlay',
+});
+const FLYOUT_TYPE_PUSH_MODE = i18n.translate('expandableFlyout.renderMenu.pushMode', {
+  defaultMessage: 'Push',
+});
+
+interface RenderMenuProps {
+  /**
+   * Current flyout type
+   */
+  flyoutTypeProps: {
+    /**
+     * 'push' or 'overlay'
+     */
+    type: EuiFlyoutProps['type'];
+    /**
+     * Callback to change the flyout type
+     */
+    onChange: (type: EuiFlyoutProps['type']) => void;
+    /**
+     * Disables the button group for flyout where the option shouldn't be available
+     */
+    disabled: boolean;
+    /**
+     * Allows to show a tooltip to explain why the option is disabled
+     */
+    tooltip: string;
+  };
+}
+
+/**
+ * Renders a menu to allow the user to customize the flyout.
+ * Current customization are:
+ * - Flyout type: overlay or push
+ */
+export const RenderMenu: React.FC<RenderMenuProps> = ({ flyoutTypeProps }: RenderMenuProps) => {
+  const [isPopoverOpen, setPopover] = useState(false);
+  const togglePopover = () => {
+    setPopover(!isPopoverOpen);
+  };
+
+  const panels = [
+    {
+      id: 0,
+      title: 'Layout settings',
+      content: (
+        <EuiPanel paddingSize="s">
+          <EuiTitle size="xxs" data-test-subj={RENDER_MENU_FLYOUT_TYPE_TITLE_TEST_ID}>
+            <h3>
+              {FLYOUT_TYPE_TITLE}{' '}
+              {flyoutTypeProps.tooltip && (
+                <EuiToolTip position="top" content={flyoutTypeProps.tooltip}>
+                  <EuiIcon
+                    data-test-subj={RENDER_MENU_FLYOUT_TYPE_INFORMATION_ICON_TEST_ID}
+                    type="iInCircle"
+                    css={css`
+                      margin-left: 4px;
+                    `}
+                  />
+                </EuiToolTip>
+              )}
+            </h3>
+          </EuiTitle>
+          <EuiSpacer size="s" />
+          <EuiButtonGroup
+            legend={FLYOUT_TYPE_TITLE}
+            options={[
+              {
+                id: 'overlay',
+                label: FLYOUT_TYPE_OVERLAY_MODE,
+                'data-test-subj': RENDER_MENU_FLYOUT_TYPE_BUTTON_GROUP_OVERLAY_TEST_ID,
+                toolTipContent: 'Shows the flyout on top of the page content',
```

Comment:
@PhilippeOberti how's this:

```suggestion
                toolTipContent: 'Displays the flyout over the page',
```

Reply:
that works for me! Thank you so much for the quick feedback!